tcms/testcases/static/testcases/js/duration_picker.js

```diff
@@ -0,0 +1,56 @@
+  $(function () {
```

Comment:
This looks like a JavaScript library and should be installed as such via package.json, not vendored in into the repository.

More over if you are taking source code from somebody else, there are issues with licensing and copyright. You can't just commit this under your own name so please explain what this is.

Reply:
I was using this [bootstrap reference](https://github.com/koss-lebedev/bootstrap-duration-picker) but was not sure how to plug in the feature because it has the JavaScript code that gets plugged. There is also this [demo](https://github.com/koss-lebedev/bootstrap-duration-picker/blob/master/demo/example.html) in that library that I used. 

They also have the [license](https://github.com/koss-lebedev/bootstrap-duration-picker/blob/master/LICENSE) included there that says the references could be copied to use the library

Reply:
See https://docs.djangoproject.com/en/3.1/howto/static-files/ & `tcms/package.json`

Reply:
Also https://docs.djangoproject.com/en/3.1/ref/forms/widgets/ + tcms/core/widgets.py